### PR TITLE
Add some ICEs

### PR DESCRIPTION
**Pull request by fanninpm**

### CODE REVIEW

File: ices/83394.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+rustc -C embed-bitcode=no -C codegen-units=1 -C debuginfo=2 --crate-type lib - <<'EOF'
```
Comment (JohnTitor) [inline ices/83394.sh:3]: Why don't you just use `#![crate_type]` here?

**Review by JohnTitor — CHANGES_REQUESTED** [review]

File: ices/83394.sh (hunk 1 above)
Comment (fanninpm) [reply ices/83394.sh]: Resolved in ff2dfb5.

File: ices/83394.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+struct Integer(u8);
+
+
+// triggers different compiler panic
+/*
+#[repr(C)]
+#[derive(Debug)]
+#[rustc_layout_scalar_valid_range_end(0xFFFF)]
+struct Integers(u8, u8);
+*/
```
Comment (JohnTitor) [inline ices/83394.rs:17]:
```suggestion
*/

```

**Comment by JohnTitor** [comment]
Failures are unrelated, thanks!

**Pull request merged**